### ISSUE TITLE
OutcomesController: Error after removing child state

### ISSUE THREAD

**Issue by franzlst** [DLR-RM/RAFCON#228]
The OutcomesController of the root state logs the following errors, when a port is selected after removing a child state:

```
13:50:47 ERROR - mvc.models.state_machine: path element HEOLJT of '['HEOLJT', 'YPLQUN']' could not be found root_state is new root state HEOLJT
13:50:47 ERROR - mvc.models.state_machine: path element HEOLJT of '['HEOLJT', 'YPLQUN']' could not be found root_state is new root state HEOLJT`
```

Steps to reproduce:
1. Create new state machine
2. Add child state
3. Remove child state (select + <Del>)
4. Select root state
5. Select any outcome within the OutcomemsController

Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2016-11-17 12:53:22+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
So respective outcomes widget of the removed state is not destroyed. So the failure seems to be a general problem of the states editor and know memory leak of RAFCON. I improved the transitions and data flows widgets to be robust against it. Anyway every widget it self has to be robust against such problems (because the widget can be used stand alone or some where else in future). 

The issue stays open and I will fix it. I will open a issue for the states editor that will include a sub-task for a unit test on remove as well widget destroy functionality.


Originally created by @Rbelder  at 2016-11-17 14:55:55+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
My comment before was not fully right. The outcomes, transitions and data flows widgets have a different architecture then the other widgets. This maybe should and can be corrected what will be part of issue #229. This issue can be closed as long as the destruction of this widget works fine with the relieve of all models what was missing, see commit 23c6b10.


Originally created by @Rbelder  at 2016-11-17 17:09:34+00:00 (moved from RMC internal repository)
